### PR TITLE
HDDS-7076. Log container file path when container cannot be written.

### PR DESCRIPTION
**Pull request by Galsza**
## What changes were proposed in this pull request?

Add container file path to error message when writing to container encounters an error

## What is the link to the Apache JIRA

[HDDS-7076](https://issues.apache.org/jira/browse/HDDS-7076)

## How was this patch tested?

Reproduce the error on a local cluster via taking away the write permission on container data by making its folder read only.

### CODE REVIEW

File: hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/keyvalue/KeyValueContainer.java [hunk 1]
```diff
@@ -260,7 +260,8 @@ private void writeToContainerFile(File containerFile, boolean isCreate)
     } catch (IOException ex) {
       onFailure(containerData.getVolume());
       throw new StorageContainerException("Error while creating/ updating " +
-          ".container file. ContainerID: " + containerId, ex,
+          ".container file. ContainerID: " + containerId + " Container path: " +
```
Comment (duongnguyen0) [inline hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/keyvalue/KeyValueContainer.java:263]:
```suggestion
          " container file. ContainerID: " + containerId + ", container path: " +
```

Comment (errose28) [reply hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/keyvalue/KeyValueContainer.java:263]: I think the `.` was originally there to signify the file extension of the file is `.container`. I don't think it matters either way, just a note.

Comment (Galsza) [reply hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/keyvalue/KeyValueContainer.java]: Thanks for spotting this, I've fixed it.

File: hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/keyvalue/KeyValueContainer.java [hunk 2]
```diff
@@ -260,7 +260,8 @@ private void writeToContainerFile(File containerFile, boolean isCreate)
     } catch (IOException ex) {
       onFailure(containerData.getVolume());
       throw new StorageContainerException("Error while creating/ updating " +
-          ".container file. ContainerID: " + containerId, ex,
+          " container file. ContainerID: " + containerId +
+          " ,container path: " + containerFile.getAbsolutePath(), ex,
```
Comment (adoroszlai) [inline hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/keyvalue/KeyValueContainer.java:264]: Note that error may happen for temp file (lines 246-248) or final file (lines 254, 256-257).  Can we distinguish between those two cases?

Also, spacing is a bit odd here (some double, some missing).  How about:

```java
      throw new StorageContainerException("Error while creating/updating" +
          " container file. ContainerID: " + containerId +
          ", container path: " + containerFile.getAbsolutePath(), ex,
```

Comment (Galsza) [reply hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/keyvalue/KeyValueContainer.java]: Added additional text when the temp file could not be created. I'm not sure what else we can do with this situation.

**Review by adoroszlai — APPROVED** [review]

**Comment by adoroszlai** [comment]
Thanks @Galsza for updating the patch.  Thanks @duongnguyen0 for the review.

**Pull request merged**